VmAgent.Core.UnitTets/SystemOperationsTests.cs

```diff
@@ -0,0 +1,293 @@
+            _mockFileSystemOperations.Setup(x => x.TryGetParentDirectory(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _subdirectoryPath)), out _directoryInfo)).Returns(true);
+            _mockFileSystemOperations.Setup(x => x.Exists(It.Is<FileSystemInfo>((info) => IsFileSystemInfoEqual((DirectoryInfo)info, _root)))).Returns(true);
+            _mockFileSystemOperations.Setup(x => x.Exists(It.IsAny<FileSystemInfo>())).Returns(false);
+        }
+
+        bool IsFileSystemInfoEqual(FileSystemInfo toCompare, string path)
+        {
+            return toCompare.FullName == path;
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestSetUnixOwnerRecursive()
+        {
+            List<string> expectedItemsToOwn = new List<string>()
+            {
+                _directoryPath,
+                _subdirectoryPath,
+                _directoryFilePath,
+                _subdirectoryFilePath
+            };
+
+            _systemOperations.SetUnixOwnerIfNeeded(_directoryPath, true);
+
+            expectedItemsToOwn.ForEach(
+                    (item) => _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path == item), _systemOperations.User), Times.Once));
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestSetUnixOwnerNonRecursive()
+        {
+            List<string> unExpectedItemsToOwn = new List<string>()
+            {
+                _subdirectoryPath,
+                _directoryFilePath,
+                _subdirectoryFilePath
+            };
+
+            _systemOperations.SetUnixOwnerIfNeeded(_directoryPath, false);
+
+            unExpectedItemsToOwn.ForEach(
+                    (item) => _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path == item), _systemOperations.User), Times.Never));
+            _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.Is<string>((path) => path.EndsWith(_directoryPath)), _systemOperations.User), Times.Once);
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
+        public void TestSetUnixOwnerDoesNothing()
+        {
+            List<string> expectedItemsToOwn = new List<string>()
+            {
+                _directoryPath,
+                _subdirectoryPath,
+                _directoryFilePath,
+                _subdirectoryFilePath
+            };
+            VmConfiguration config = new VmConfiguration(56001, "vmid", new VmDirectories("root"), false);
+            SystemOperations systemOperations = new SystemOperations(config, _logger, _mockFileSystemOperations.Object);
+            systemOperations.SetUnixOwnerIfNeeded(_directoryPath, true);
+
+            expectedItemsToOwn.ForEach(
+                    (item) => _mockFileSystemOperations.Verify(x => x.SetUnixOwner(It.IsAny<string>(), It.IsAny<string>()), Times.Never));
+        }
+
+        [TestMethod]
+        [TestCategory("BVT")]
```

Comment:
Since this is only for unit tests, you can place the  [TestCategory("BVT")] at the class level. I dont see any other types of tests here?

Reply:
All our tests are done this way, I find it helps to make explicit the category when looking at the test

Reply:
But this is only a unit test file ( its under UnitTests.)

Reply:
This is a unit test class though, there are no other categories is what i meant.